### DISCUSSION TITLE
Animation for sprites.

### DISCUSSION THREAD

**Discussion by Phil-hacker** [godotengine/godot-proposals#4726] [2D]
When using the 2DSprite you can set animation frames but you have to write a script to play the animation. The solution for that would be an fps property that would play the animation automatically. 
You can achieve this with a simple script but it would be better if it would be an integrated feature.

**Comment by Mickeon** [comment]
Or you can use AnimatedSprite?

**Comment by Phil-hacker** [comment]
But you can't use an atlas in the AnimatedSprite.
AnimatedSprites are also way more complicated, because you have to create a SpriteFrames resource.

**Reply by Calinou** [reply]
AnimatedSprite is intended to be used for basic use cases. For advanced use cases, Sprite + AnimationPlayer will always remain a more powerful and flexible combination.

**Reply by dalexeev** [reply]
I think `AnimatedSprite`/`SpriteFrames` should be improved. At least, we should add the ability [to configure individual delays between frames](https://github.com/godotengine/godot-proposals/issues/2214), but other improvements are probably possible. `AnimationPlayer` is universal, but IMO it's less convenient and takes more time to set up animations.

**Comment by 1Luc1** [comment]
I'm fairly new to godot, but i used the Sprite2D with the AnimationPlayer; they really work well together.

See this nice tutorial here -> [YouTube.Link](https://youtu.be/wX145eoLFSM)
